### PR TITLE
Update Windows CI to use gcc-10

### PR DESCRIPTION
**Pull request by johnplate**
Update Windows CI to use gcc-10 to be in sync with [#570](https://github.com/google/agi/pull/570).

### CODE REVIEW

File: kokoro/windows/build.bat [hunk 1]
```diff
@@ -39,32 +39,28 @@ wget -q https://dl.google.com/android/repository/android-ndk-r21d-windows-x86_64
...
+.\msys2-base-x86_64-20201109.sfx.exe -y -o%BUILD_ROOT%\
+
+REM Start empty shell to initialize MSYS2.
+%BUILD_ROOT%\msys64\usr\bin\bash -lc " "
+
+REM Uncomment the following line to list all packages and versions installed in MSYS2.
+REM %BUILD_ROOT%\msys64\usr\bin\bash -lc "pacman -Q"
+
+REM Install packages required by the build process.
+%BUILD_ROOT%\msys64\usr\bin\bash -lc "pacman -S --noconfirm git patch zip unzip"
```
Comment (hevrard) [inline kokoro/windows/build.bat:54]: I think we should install **all** packages from known-to-work fixed version, as we do for e.g. gcc. In the past we had issues with e.g. a new version of `patch` that broke our build.

So I suggest using the same approach as gcc for git, patch, zip and unzip.

**Review by hevrard — COMMENTED** [review]
Some of the changes are already present in #570

File: kokoro/windows/build.bat (hunk 1 above)
Comment (johnplate) [reply kokoro/windows/build.bat]: Thanks, done

**Comment by johnplate** [comment]
> Some of the changes are already present in #570

Yes, this PR is WIP until rebased after #570 is merged.

**Review by hevrard — APPROVED** [review]

File: kokoro/windows/build.bat [hunk 2]
```diff
@@ -39,32 +39,32 @@ wget -q https://dl.google.com/android/repository/android-ndk-r21d-windows-x86_64
...
-c:\tools\msys64\usr\bin\bash --login -c "pacman -U --noconfirm /t/src/mingw-w64-x86_64-crt-git-8.0.0.5647.1fe2e62e-1-any.pkg.tar.xz"
-set PATH=c:\tools\msys64\mingw64\bin;c:\tools\msys64\usr\bin;%PATH%
-set BAZEL_SH=c:\tools\msys64\usr\bin\bash
+REM Download and install MSYS2, because the pre-installed version is too old.
+REM Do NOT do a system update (pacman -Syu) because it is a moving target.
+wget -q https://github.com/msys2/msys2-installer/releases/download/2020-11-09/msys2-base-x86_64-20201109.sfx.exe
+.\msys2-base-x86_64-20201109.sfx.exe -y -o%BUILD_ROOT%\
+
+REM Start empty shell to initialize MSYS2.
+%BUILD_ROOT%\msys64\usr\bin\bash -lc " "
```
Comment (pmuetschard) [inline kokoro/windows/build.bat:48]: keep it as `--login -c`. For most readers `--login` is easier to understand than `-l`.

**Review by pmuetschard — APPROVED** [review]

File: kokoro/windows/build.bat (hunk 2 above)
Comment (johnplate) [reply kokoro/windows/build.bat]: Sure, done

**Pull request merged**